Landry333/Big-Owl
# #15 user sms alert if user location cannot be detected

Related to issue #15

**Description**
The feature in this PR will check 4 min after the beginning of a schedule if there is any user for the schedule who has not been detected for location by the system. For all such users, the system will send them a text SMS message from the supervisor phone to warn these users to check their setting for location detection. 
For every current user profile opened on the phone, the feature checks the next coming schedules for which the user profile is a supervisor and creates an alarm 4 min after the start time of every schedule. The alarm will trigger a check in the corresponding schedule for schedule users whose "scheduleLocated" status is still "NOT_LOCATED" and will send them a warning text SMS. The alarms are set each time you visit the homepage.

**How to check PR**

You can use two phone numbers as in real conditions or test it with only one phone number. 

- Open a supervisor profile on your phone, make sure you have accepted or enable the SMS permissions and create a schedule for a user / worker
- Make sure the app can't detect the user / worker location (that is the app should not be working normally on the worker's phone so can't detect the user / worker location)
- 4 min after the beginning of the schedule, the app on the supervisor phone will send a warning SMS text message to the worker's phone (here you can put your number [supervisor] on firebase as the worker's phone number to send the message to yourself )

You can also test with multiple schedule users / workers
